src/xform/barrier_analysis.rs

```diff
@@ -0,0 +1,1297 @@
+                          .get(&expr.get_operand_value(0).as_ref().unwrap().get_key())
+                          .unwrap(),
+                        start_node,
+                        end_node,
+                      );
+                      new_expr_handle = Some(new_expr);
+                    }
+                  }
+                }
+                Opcode::Compare { cmp } => match cmp {
+                  expr::subcode::Compare::IGT => {
+                    let new_expr = self.sys.create_igt(
+                      *node_remapping_map
+                        .get(&expr.get_operand_value(0).as_ref().unwrap().get_key())
+                        .unwrap(),
+                      *node_remapping_map
+                        .get(&expr.get_operand_value(1).as_ref().unwrap().get_key())
+                        .unwrap(),
+                    );
+                    new_expr_handle = Some(new_expr);
+                  }
+                  expr::subcode::Compare::IGE => {
+                    let new_expr = self.sys.create_ige(
+                      *node_remapping_map
+                        .get(&expr.get_operand_value(0).as_ref().unwrap().get_key())
+                        .unwrap(),
+                      *node_remapping_map
+                        .get(&expr.get_operand_value(1).as_ref().unwrap().get_key())
+                        .unwrap(),
+                    );
+                    new_expr_handle = Some(new_expr);
+                  }
+                  expr::subcode::Compare::ILT => {
+                    let new_expr = self.sys.create_ilt(
+                      *node_remapping_map
+                        .get(&expr.get_operand_value(0).as_ref().unwrap().get_key())
+                        .unwrap(),
+                      *node_remapping_map
+                        .get(&expr.get_operand_value(1).as_ref().unwrap().get_key())
+                        .unwrap(),
+                    );
+                    new_expr_handle = Some(new_expr);
+                  }
+                  expr::subcode::Compare::ILE => {
+                    let new_expr = self.sys.create_ile(
+                      *node_remapping_map
+                        .get(&expr.get_operand_value(0).as_ref().unwrap().get_key())
+                        .unwrap(),
+                      *node_remapping_map
+                        .get(&expr.get_operand_value(1).as_ref().unwrap().get_key())
+                        .unwrap(),
+                    );
+                    new_expr_handle = Some(new_expr);
+                  }
+                  expr::subcode::Compare::EQ => {
+                    let new_expr = self.sys.create_eq(
+                      *node_remapping_map
+                        .get(&expr.get_operand_value(0).as_ref().unwrap().get_key())
+                        .unwrap(),
+                      *node_remapping_map
+                        .get(&expr.get_operand_value(1).as_ref().unwrap().get_key())
+                        .unwrap(),
```

Comment:
Consider to merge these instructions, please refer to `instructions.rs` for more concise read-only usages.